XenAdmin/SettingsPanels/CPUMemoryEditPage.cs

```diff
@@ -220,39 +214,106 @@ public void Repopulate()
                 lblVcpuWarning.Visible = false;
             }
 
+            isVcpuHotplugSupported = vm.SupportsVcpuHotplug;
+
+            label1.Text = GetRubric();
+
             _OrigMemory = nudMemory.Value;
-            _OrigVCPUs = vm.VCPUs_at_startup > 0 ? vm.VCPUs_at_startup : 1;
+            _OrigVCPUsMax = vm.VCPUs_max > 0 ? vm.VCPUs_max : 1;
+            _OrigVCPUsAtStartup = vm.VCPUs_at_startup > 0 ? vm.VCPUs_at_startup : 1;
             _OrigVCPUWeight = _CurrentVCPUWeight;
+            // _OrigVCPUs represents VCPUs_max if hotplug is allowed, otherwise VCPUs_at_startup

```

Comment:
I think it would be worth pulling this comment out to the top of the file and adding a bit more explanation. Including why there are three, and what the two combo boxes represent in different situations.